lib/src/rules/avoid_types_as_parameter_names.dart

```diff
@@ -60,13 +60,22 @@ class _Visitor extends SimpleAstVisitor<void> {
   @override
   void visitFormalParameterList(FormalParameterList node) {
     if (node.parent is GenericFunctionType) return;
+    _visitParameters(node, node.parameters);
+  }
 
-    for (var parameter in node.parameters) {
+  void _visitParameters(
+      FormalParameterList node, NodeList<FormalParameter> parameters) {
+    for (var parameter in parameters) {
       var declaredElement = parameter.declaredElement;
       var identifier = parameter.identifier;
+      if (parameter is SimpleFormalParameter) {
+        var type = parameter.type;
+        if (type is GenericFunctionType) {
```

Comment:
We don't need to do this. If the type of the parameter is a `GenericFunctionType` then we'll visit the formal parameters of that type independently (as part of visiting the whole AST). I believe that you can remove lines 71-76 without breaking anything (and having them means that we'll get more than one lint reported, which is undesirable).

If we don't already have one it would be good to add a test to confirm that this case is covered, perhaps something like:
```
void f(int Function<T>(String T) g) {} // LINT
```

If you delete those lines then I don't think there's any value in splitting out this separate method, so consider reverting those changes as well.

Reply:
You are right, I should have thought a little about this. Reverted.